src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+    {
+      name: 'previousScreen',
+      expression: function(wizardPosition) {
+        return this.nextAvailable(wizardPosition, this.positionBefore.bind(this));
+      }
+    },
+    {
+      name: 'visitedWizardlets',
+      class: 'FObjectArray',
+      of: 'foam.u2.wizard.Wizardlet'
+    },
+
+    // Convenience properties
+    {
+      name: 'currentWizardlet',
+      expression: function(wizardlets, wizardPosition) {
+        var current = wizardlets[wizardPosition.wizardletIndex];
+        if ( this.visitedWizardlets.indexOf(current) == -1 ) {
+          this.visitedWizardlets.push(current);
+        }
+        return current;
+      }
+    },
+    {
+      name: 'currentSection',
+      expression: function(currentWizardlet, wizardPosition$sectionIndex) {
+        return this.currentWizardlet.currentSection = currentWizardlet
+            .sections[wizardPosition$sectionIndex];
+      }
+    },
+    {
+      name: 'isLastScreen',
+      class: 'Boolean',
+      expression: function(nextScreen) {
+        return nextScreen == null;
+      }
+    },
+    {
+      name: 'canGoBack',
+      class: 'Boolean',
+      expression: function(previousScreen) {
+        return previousScreen != null;
+      }
+    },
+    {
+      name: 'canGoNext',
+      class: 'Boolean',
+      expression: function(
+        wizardPosition, nextScreen,
+        currentSection, currentWizardlet,
+        currentSection$isValid, currentWizardlet$isValid,
+        config$allowSkipping
+      ) {
+        if ( config$allowSkipping ) return true;
+        if (
+          ! nextScreen ||
+          wizardPosition.wizardletIndex != nextScreen.wizardletIndex
+        ) {
+          if ( ! currentWizardlet$isValid ) return false;
+        }
+        return currentSection$isValid;
+      }
+    },
+    {
+      name: 'submitted',
+      class: 'Boolean'
+    }
+  ],
+
+  methods: [
+    function setupWizardletListeners(wizardlets) {
+      console.log('step wizard', this, wizardlets);
+      if ( this.wsub ) this.wsub.detach();
+      this.wsub = this.FObject.create();
+
+      var wi = 0, si = 0;
+      wizardlets.forEach((w, wizardletIndex) => {
+
+        // Bind availability listener for wizardlet availability
+        var isAvailable$ = w.isAvailable$;
+        console.log(wizardletIndex, w, w.isAvailable$, isAvailable$);
+        this.wsub.onDetach(isAvailable$.sub(() => {
+          console.log('X');
+          this.onWizardletAvailability(wizardletIndex, isAvailable$.get());
+        }));
+
+        // Bind availability listener for each wizardlet section
+        w.sections.forEach((section, sectionIndex) => {
+          var sectionPosition = this.WizardPosition.create({
+            wizardletIndex: wizardletIndex,
+            sectionIndex: sectionIndex,
+          });
+          var isAvailable$ = section.isAvailable$;
+          this.wsub.onDetach(isAvailable$.sub(() => {
+            console.log('Y');
+            this.onWizardPositionAvailabilityUpdate(
```

Comment:
are you sure you have the right function here... shouldn't it be 'onSectionAvailability'

makes me wonder why are you creating a new WizardPosition here. Shouldn't their only be one wizardPosition?

Reply:
> are you sure you have the right function here... shouldn't it be 'onSectionAvailability'
fixed, thanks

> makes me wonder why are you creating a new WizardPosition here. Shouldn't their only be one wizardPosition?
I'm not aware of any reason there should only be one instance of WizardPosition so this question is confusing to me. WizardPosition objects can be compared and having multiple objects to represent different positions is useful.